From 96aa7d6a55e2a9b580649294c7f8b2b306ac7d20 Mon Sep 17 00:00:00 2001
From: Xiaolei Wang <xiaolei.wang@windriver.com>
Date: Tue, 3 Mar 2020 20:49:42 +0800
Subject: [PATCH] drm/imx: lcdif: fix build error

When we enable CONFIG_IMX_LCDIF_CORE driver, the build error occurs.
drm_fb_cma_create has been deleted we can use drm_gem_fb_create
drm_plane_helper_check_state has been deleted we can use
drm_atomic_helper_check_plane_state

Signed-off-by: Xiaolei Wang <xiaolei.wang@windriver.com>
---
 drivers/gpu/drm/imx/lcdif/lcdif-kms.c   | 4 ++--
 drivers/gpu/drm/imx/lcdif/lcdif-plane.c | 2 +-
 2 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/drivers/gpu/drm/imx/lcdif/lcdif-kms.c b/drivers/gpu/drm/imx/lcdif/lcdif-kms.c
index f013bf0ce7f8..0b0e494c8981 100644
--- a/drivers/gpu/drm/imx/lcdif/lcdif-kms.c
+++ b/drivers/gpu/drm/imx/lcdif/lcdif-kms.c
@@ -17,7 +17,7 @@
 #include <drm/drm_atomic_helper.h>
 #include <drm/drm_crtc.h>
 #include <drm/drm_fb_cma_helper.h>
-
+#include <drm/drm_gem_framebuffer_helper.h>
 static void lcdif_drm_atomic_commit_tail(struct drm_atomic_state *state)
 {
 	struct drm_device *dev = state->dev;
@@ -36,7 +36,7 @@ static void lcdif_drm_atomic_commit_tail(struct drm_atomic_state *state)
 }
 
 const struct drm_mode_config_funcs lcdif_drm_mode_config_funcs = {
-	.fb_create     = drm_fb_cma_create,
+	.fb_create     = drm_gem_fb_create,
 	.atomic_check  = drm_atomic_helper_check,
 	.atomic_commit = drm_atomic_helper_commit,
 };
diff --git a/drivers/gpu/drm/imx/lcdif/lcdif-plane.c b/drivers/gpu/drm/imx/lcdif/lcdif-plane.c
index a29fd3ddc6f5..587d257ab5dd 100644
--- a/drivers/gpu/drm/imx/lcdif/lcdif-plane.c
+++ b/drivers/gpu/drm/imx/lcdif/lcdif-plane.c
@@ -71,7 +71,7 @@ static int lcdif_plane_atomic_check(struct drm_plane *plane,
 	clip.x2 = mode->hdisplay;
 	clip.y2 = mode->vdisplay;
 
-	ret = drm_plane_helper_check_state(plane_state, &clip,
+	ret = drm_atomic_helper_check_plane_state(plane_state, &clip,
 					   DRM_PLANE_HELPER_NO_SCALING,
 					   DRM_PLANE_HELPER_NO_SCALING,
 					   false, true);
-- 
2.17.1

